src/lib/homestore_backend/snapshot_receive_handler.cpp

```diff
@@ -59,13 +58,14 @@ int HSHomeObject::SnapshotReceiveHandler::process_shard_snapshot_data(ResyncShar
     shard_sb.chunk_id = chunk_id;
 
     homestore::MultiBlkId blk_id;
-    const auto hints = home_obj_.chunk_selector()->chunk_to_hints(chunk_id);
+    const auto hints = home_obj_.chunk_selector()->chunk_to_hints(shard_sb.chunk_id);
     auto status = homestore::data_service().alloc_blks(
         sisl::round_up(sizeof(shard_sb), homestore::data_service().get_blk_size()), hints, blk_id);
     if (status != homestore::BlkAllocStatus::SUCCESS) {
         LOGE("Failed to allocate blocks for shard {}", shard_meta.shard_id());
         return ALLOC_BLK_ERR;
     }
+    shard_sb.chunk_id = blk_id.to_single_blkid().chunk_num(); // FIXME: remove this after intergating vchunk
 
     const auto ret = homestore::data_service()
```

Comment:
NIT: maybe we can use `async_alloc_write` to combine line 62 and line 70

Reply:
Good suggestion, but seems it need several extra lines to build the input sg_list from buf. Will check later if there's any simplier way to do it.